docs/customization_extendibility.md

```diff
@@ -0,0 +1,72 @@
+# Extending and Customizing
+For many simpler web applications, `ddev` provides everything you need to successfully provision and develop a web application on your local machine out of the box. More complex and sophisticated web applications, however, often require integration with services beyond the standard requirements of a web and database server. Examples of these additional services are Apache Solr, Redis, Varnish, etc. While `ddev` likely won't ever provide all of these additional services out of the box, it is designed to provide simple ways for the environment to be customized and extended to meet the needs of your application.
+
+## Prerequisite knowledge
+The majority of `ddev`'s customization ability and extensibility comes from leveraging features and functionality provided by [Docker](https://docs.docker.com/) and [Docker Compose](https://docs.docker.com/compose/overview/). Some working knowledge of these tools is required in order to customize or extend the environment `ddev` provides.
```

Comment:
Although we've been doing it everywhere, I don't think we should put ddev in single quotes except when demonstrating a command. It just looks weird, as you see on https://github.com/tannerjfco/ddev/blob/690241b3530d373900a5f79290f9db37db4bb1c2/docs/customization_extendibility.md